packages/material-ui/src/useWidth/unstable_useWidth.test.js

```diff
@@ -0,0 +1,109 @@
+/* eslint-disable react/prop-types */
+import React from 'react';
+import { assert } from 'chai';
+import useWidth from './unstable_useWidth';
+import { createMount } from '@material-ui/core/test-utils';
+
+function TestWidth({ widthParam }) {
+  const { width } = useWidth({ width: widthParam });
+  return <div id="result">{width}</div>;
+}
+
+function TestIsWidthUp({ breakpoint, inclusive }) {
+  const { isWidthUp } = useWidth();
+  return <div id="result">{`${isWidthUp(breakpoint, inclusive)}`}</div>;
+}
+
+function TestIsWidthDown({ breakpoint, inclusive }) {
+  const { isWidthDown } = useWidth();
+  return <div id="result">{`${isWidthDown(breakpoint, inclusive)}`}</div>;
+}
+
+const TEST_WIDTH = 300;
+
+const changeWindowWidth = width => {
+  window.innerWidth = width;
+};
+
+describe('useWidth()', () => {
+  let mount;
+  let originalWidth;
+
+  before(() => {
+    mount = createMount();
+    originalWidth = window.innerWidth;
+  });
+
+  after(() => {
+    mount.cleanUp();
+  });
+
+  afterEach(() => {
+    // restore original window width
+    changeWindowWidth(originalWidth);
+  });
+
+  describe('width', () => {
+    it('returns correct width', () => {
+      // custom window dimensions
+      changeWindowWidth(TEST_WIDTH);
+      const wrapper = mount(<TestWidth />);
+
+      assert.strictEqual(wrapper.find('#result').text(), 'xs');
+    });
+  });
+
+  describe('isWidthUp', () => {
+    it('should work as default inclusive', () => {
+      changeWindowWidth(1024);
+      const largeWrapper = mount(<TestIsWidthUp breakpoint="lg" />);
+      const mediumWrapper = mount(<TestIsWidthUp breakpoint="md" />);
+      const smallWrapper = mount(<TestIsWidthUp breakpoint="sm" />);
+
+      assert.strictEqual(largeWrapper.find('#result').text(), 'false', 'should accept larger size');
```

Comment:
We don't really that selector. `#text` on the wrapper should return the same result.